xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -430,6 +430,20 @@ func (b *cdsBalancer) ExitIdle() {
 	})
 }
 
+// Node ID needs to be manually added to errors generated in the following
+// scenarios:
+//   - resource-does-not-exist: since the xDS watch API uses a separate callback
+//     instead of returning an error value,
```

Comment:
This goes away later, right?  Maybe TODO here since we know it might become outdated?

Reply:
Yes.

Added todo.